frontend/components/MapMicro.js

```diff
@@ -256,14 +443,16 @@ export default class MapMicro extends React.Component {
     };
 
     render() {
-        const currentMarkerObjects = this.state.markerData.filter((location) => (
-            location.year >= this.state.lastValid[0] && location.year <= this.state.lastValid[1])
-        );
+        const validAddresses = this.state.markerData.filter((location) => (
```

Comment:
This will get refiltered and mapped on every component rerender. Best to move this to componentDidMount